test/src/main/configurations/MainConfig/ConfigurationSoapValidator.xml

```diff
@@ -0,0 +1,42 @@
+
+<adapter name="SoapValidator">
+	<receiver className="nl.nn.adapterframework.receivers.GenericReceiver" name="SoapValidator">
+		<listener className="nl.nn.adapterframework.receivers.JavaListener" serviceName="ibis4test-SoapValidator" throwException="false" />
+	</receiver>
+	<pipeline firstPipe="HelloWorld">
+		<exits>
+			<exit path="EXIT" state="success" />
+		</exits>
+		<inputValidator className="nl.nn.adapterframework.soap.SoapValidator" schemaLocation="http://api.nn.nl/HelloWorld /SoapValidator/xsd/HelloWorld.xsd" soapBody="HelloWorld_Request" addNamespaceToSchema="true">
+			<forward name="failure" path="InputValidatorFailure" />
+		</inputValidator>
+		<inputWrapper className="nl.nn.adapterframework.extensions.api.ApiSoapWrapperPipe" direction="unwrap" />
+		<outputWrapper className="nl.nn.adapterframework.extensions.api.ApiSoapWrapperPipe" outputNamespace="http://api.nn.nl/HelloWorld" root="HelloWorld_Response" />
+		<outputValidator className="nl.nn.adapterframework.soap.SoapValidator" schemaLocation="http://api.nn.nl/HelloWorld /SoapValidator/xsd/HelloWorld.xsd" soapBody="HelloWorld_Response" addNamespaceToSchema="true">
+			<forward name="failure" path="OutputValidatorFailure" />
+		</outputValidator>
+		<pipe name="InputValidatorFailure" className="nl.nn.adapterframework.pipes.XsltPipe" styleSheetName="xml/xsl/esb/soapBody.xsl" xslt2="true" getInputFromSessionKey="xmlFailureReason" storeReplacedInputInSessionKey="replacedInput">
+			<param name="cmhVersion" value="1" />
+			<param name="errorCode" value="ERR6003" />
+			<param name="errorReason" value="Invalid Request" />
+			<param name="errorRootTag" value="HelloWorld_Response" />
+			<param name="originalMessage" sessionKey="replacedInput" />
+			<param name="addErrorsFromInputMessage" value="true" />
+			<param name="operationName" value="Test/SoapValidator" />
+			<param name="paradigm" value="Response" />
+			<forward name="success" path="EXIT" />
+		</pipe>
```

Comment:
Hier gebruik je storeReplacedInputInSessionKey i.c.m. getInputFromSessionKey="xmlFailureReason".
Vervolgens gebruik je de originele input als parameter.
Dat kan natuurlijk ook andersom, dat je de orignele input de input laat zijn, en de failure reason als parameter.
Dan is storeReplacedInputInSessionKey niet nodig.

Reply:
Aangepast zoals door jou voorgesteld